tests/test_add_news.py

```diff
@@ -15,23 +14,24 @@ def _setup_news_test_env(tmp_path):
     project_root = Path(__file__).resolve().parents[1]
     real_template_path = project_root / "news" / "TEMPLATE.rst"
     test_template_file = test_news_dir / "TEMPLATE.rst"
-    # Copy the real template to the test directory
     shutil.copy(real_template_path, test_template_file)
-    # Override paths for testing
-    add.NEWS_DIR = str(test_news_dir)
-    add.TEMPLATE_PATH = str(test_template_file)
-    # Mock branch setup
+    # Mock the paths and the branch
+    mocker.patch("scikit_package.cli.add.NEWS_DIR", str(test_news_dir))
+    mocker.patch(
+        "scikit_package.cli.add.TEMPLATE_PATH", str(test_template_file)
+    )
     branch_name = "test-branch"
-    import scikit_package.utils.auth as auth
-
-    auth.get_current_branch = lambda: branch_name
+    mocker.patch(
+        "scikit_package.utils.auth.get_current_branch",
```

Comment:
Override the function of getting the currently checked-out branch to "test-branch" with mocker.